File: anagram/src/main/java/com/example/kata/anagram/AnagramResolver.java
package com.example.kata.anagram;

import java.util.Set;
import java.util.stream.Collectors;

class AnagramResolver {
    private final AnagramGenerator generator;

    AnagramResolver(Set<String> wordList) {
        this.generator = new AnagramGenerator(wordList);
    }

    Set<Anagram> resolve(String source) {
        return generator.generate().filter(anagram -> anagram.match(source)).collect(Collectors.toSet());
    }
}

File: anagram.js/src/AnagramResolver.js
import AnagramGenerator from "./AnagramGenerator";

export default class AnagramResolver {
    constructor(wordList) {
        this.generator = new AnagramGenerator(wordList);
    }

    resolve(source) {
        const result = [];
        this.generator.walk(anagram => anagram.match(source) && result.push(anagram));
        return result;
    }
}